client/flutter/.gitignore

```diff
@@ -36,3 +36,6 @@ lib/generated_plugin_registrant.dart
 /coverage/
 # Exceptions to above rules.
 !/packages/flutter_tools/test/data/dart_dependencies_test/**/.packages
+
+# Clean commits
+pubspec.lock
```

Comment:
Why are you `gitignore`'ing pubspec.lock?

Reply:
`pubspec.lock` is not needed to build a Flutter app (it will be generated when running `flutter pub get`, which **needs** to be done anyway). If we do not add the file to `.gitignore`, every PR and commit will have `pubspec.lock` changes based on the environment of the committer.

I can remove it again if you say that this intended.

Reply:
I think we should include it. See https://dart.dev/guides/libraries/private-files#pubspeclock:

> For library packages, do not commit the pubspec.lock file.

> For application packages, do commit pubspec.lock. This file is the recommended way for non-shared resources such as applications to manage their dependencies.